### ISSUE TITLE
Add optional support for serde's data model

### ISSUE THREAD

**Issue by LegNeato** [rkyv/rkyv#329]
Seeing how rkyv's type system is open, it should be possible to support the serde data model on top of rkyv. This adaptor would allow integration with the serde ecosystem and could be a path to greater rkyv adoption. Once this integration exists, the serde ecosystem crates could even opt into enhanced rkyv support.

**Comment by khassar00** [comment]
I note that Flexbuffers supports serde. I hope rkyv will support serde in the future
[Flexbuffers Serde Example](https://github.com/google/flatbuffers/blob/master/samples/sample_flexbuffers_serde.rs)

**Comment by djkoloski** [comment]
Not sure exactly what's wanted here. Do you want the ability to use rkyv through serde? That's unfortunately not going to happen, the two frameworks are incompatible in ways that cannot be reconciled. Do you want the ability to use serialize serde types with rkyv? That's possible, though it might be better off as a separate crate since it's a pretty big lump of functionality that many people will not use. Do you want the ability to serialize serde types as an `ArchivedBox<[u8]>`? That's probably the domain of a wrapper type.

**Comment by LegNeato** [comment]
Yeah, that was the intention...be able to use types that impl `Serialize` / `Deserialize` without adding a feature or forking. It is my understanding that rkyv's type system can represent the serde data model, so having a "deserailizer" and a "serializer" adapter that bridges serde's data model and rkyv's type system would be very beneficial.

**Comment by styris-ame** [comment]
@djkoloski If you haven't already, take a look at how [bincode handles serde compatibility](https://docs.rs/bincode/latest/bincode/serde/index.html). It has some inherent limitations such as not supporting `#[serde(flatten)]`, but for many cases it works, and allows for serialization/deserialization of external types that implement serde serialization traits.
